### PR TITLE
General Cleanup

### PR DESCRIPTION
**Pull request by janLo**
Just general code cleanup about problems pointed out by the Clion linter.

This is based on PR #14

### CODE REVIEW

File: lib/Heartbeat/LED.cpp [hunk 1]
```diff
@@ -58,6 +58,8 @@ void LED::setState(boolean state) {
 
 boolean LED::getState() { return digitalRead(_pin) ^ !_activeHigh; }
 
+LED::~LED() {}
```
Comment (puuu) [inline lib/Heartbeat/LED.cpp]: Isn't it noise, that will be changed later again.

File: lib/Heartbeat/Heartbeat.cpp [hunk 2]
```diff
@@ -35,7 +35,7 @@ Heartbeat::Heartbeat(LED& led, int interval) : _led(led) {
   off();
 }
 
-Heartbeat::Heartbeat(int pin, int interval)
+Heartbeat::Heartbeat(uint8_t pin, unsigned int interval)
```
Comment (puuu) [inline lib/Heartbeat/Heartbeat.cpp]: This is not the file mentioned in the commit massage.

File: lib/Heartbeat/LED.cpp [hunk 3]
```diff
@@ -56,11 +58,13 @@ void LED::setState(bool state) {
   }
 }
 
-bool LED::getState() { return digitalRead(_pin) ^ !_activeHigh; }
+bool LED::getState() {
+  return static_cast<bool>(digitalRead(_pin) ^ !_activeHigh);
+}
 
-LED::~LED() {}
+LED::~LED() = default;
```
Comment (puuu) [inline lib/Heartbeat/LED.cpp]: now it is `default`. Why ever?

File: lib/Heartbeat/LED.cpp [hunk 4]
```diff
@@ -29,16 +29,18 @@
 
 #include "LED.h"
 
-LED::LED(int pin, bool activeHigh) {
+#define VAL(x) static_cast<uint8_t>((x) ? HIGH : LOW)
```
Comment (puuu) [inline lib/Heartbeat/LED.cpp]: `#define` is ugly.

File: lib/Heartbeat/LED.cpp [hunk 5]
```diff
@@ -56,11 +58,13 @@ void LED::setState(bool state) {
   }
 }
 
-bool LED::getState() { return digitalRead(_pin) ^ !_activeHigh; }
+bool LED::getState() {
+  return static_cast<bool>(digitalRead(_pin) ^ !_activeHigh);
```
Comment (puuu) [inline lib/Heartbeat/LED.cpp]: how about `digitalRead(_pin) == _activeHigh`?

File: lib/Heartbeat/LED.cpp (hunk 3 above)
Comment (janLo) [reply lib/Heartbeat/LED.cpp]: You can probably squash it together. Default is the better option.

File: lib/Heartbeat/LED.cpp (hunk 1 above)
Comment (janLo) [reply lib/Heartbeat/LED.cpp]: See https://github.com/puuu/MQTT433gateway/pull/15#discussion_r154341168

File: lib/Heartbeat/LED.cpp (hunk 4 above)
Comment (janLo) [reply lib/Heartbeat/LED.cpp]: We could just make it an inline function. For me it doesn't matter as it does not leak from the compilation unit.

File: lib/Heartbeat/LED.cpp (hunk 5 above)
Comment (janLo) [reply lib/Heartbeat/LED.cpp]: That would result in the same outcome.

File: lib/Heartbeat/Heartbeat.cpp (hunk 2 above)
Comment (janLo) [reply lib/Heartbeat/Heartbeat.cpp]: Probably an issue introduced by one of my rebases.

File: lib/Heartbeat/LED.cpp (hunk 5 above)
Comment (puuu) [reply lib/Heartbeat/LED.cpp]: but looks much better ;-)

File: lib/Heartbeat/LED.cpp (hunk 3 above)
Comment (puuu) [reply lib/Heartbeat/LED.cpp]: Lets squash.

File: lib/Heartbeat/LED.cpp (hunk 4 above)
Comment (puuu) [reply lib/Heartbeat/LED.cpp]: inline function :+1:

File: lib/Heartbeat/LED.cpp (hunk 3 above)
Comment (janLo) [reply lib/Heartbeat/LED.cpp]: done.

File: lib/Heartbeat/LED.cpp (hunk 5 above)
Comment (janLo) [reply lib/Heartbeat/LED.cpp]: done.

File: lib/Heartbeat/LED.cpp (hunk 4 above)
Comment (janLo) [reply lib/Heartbeat/LED.cpp]: done

File: lib/Heartbeat/Heartbeat.cpp (hunk 2 above)
Comment (janLo) [reply lib/Heartbeat/Heartbeat.cpp]: fixed.

**Comment by janLo** [comment]
last commit will be a seperate PR then.

**Comment by puuu** [comment]
Thanks! Merged with 9849aff2525620f7b3871b11f2c34831d862ad8d...6c9b4cf8e5678caafa9b7e7cc5cfa54de88358e4 .

**Pull request closed without merging**